### PR TITLE
Add generic error handling

### PR DESCRIPTION
**Pull request by d00rman**
This PR introduces generic error handling in service templates. It defines an _HTTPError_ object, which can be thrown by clients to signal errors. Each route handler is modified and wrapped in a promised `try()` block which allows catching sync and async errors alike. These are then handled in a generic error function set with `app.use()`.

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+
+var BBPromise = require('bluebird');
+var util = require('util');
+
+
+/**
+ * Error instance wrapping HTTP error responses
+ */
```
Comment (gwicke) [inline lib/util.js:10]: This is about the third implementation of this (others: preq, restbase). Should we export it from preq or restbase, or create a httperror package?

**Comment by gwicke** [comment]
The template seems to be gradually morphing into its own mini-framework on top of express. Should we move the core of this code into its own module in order to keep the template slim?

File: lib/util.js (hunk 1 above)
Comment (d00rman) [inline lib/util.js:10]: Heh, I was thinking about the same thing.

**Comment by d00rman** [comment]
> Should we move the core of this code into its own module in order to keep the template slim?

It does seem to be the case. How about we finish this (as I think there are not many things left, like error-handling tests), wait to hear the feedback from the mobile apps guys and then decide what/where to go (with the help of the comparison matrix we discussed today) ?

**Comment by d00rman** [comment]
Also, I have a feeling that putting our _layer_ in a separate package would steepen the learning curve for creating services. What do you think?

**Comment by gwicke** [comment]
> Also, I have a feeling that putting our layer in a separate package would steepen the learning curve for creating services. What do you think?

I think that depends on how much of the important stuff is still out in the open. Things like 'which properties are on the app object' are important to keep very clear, but some of the generic plumbing that everybody will want can IMHO be hidden in a module without any loss of clarity.

**Pull request merged**